ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/tls/TlsSpec.scala

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.tls
+
+import java.io.File
+import com.daml.bazeltools.BazelRunfiles.rlocation
+import com.daml.ledger.resources.TestResourceContext
+import com.daml.platform.apiserver.LedgerApiServer
+import io.netty.handler.ssl.ClientAuth
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AsyncWordSpec
+
+class TlsSpec
+    extends AsyncWordSpec
+    with Matchers
+    with TableDrivenPropertyChecks
+    with TestResourceContext {
+  import TlsSpec.resource
+
+  val serverCrt = resource("server.crt")
+  val serverKey = resource("server.pem")
+  val caCrt = resource("ca.crt")
+  val clientCrt = resource("client.crt")
+  val clientKey = resource("client.pem")
+
+  def tlsEnabledFixture(clientCrt: Option[File], clientKey: Option[File], clientAuth: ClientAuth) =
+    TlsFixture(tlsEnabled = true, serverCrt, serverKey, caCrt, clientCrt, clientKey, clientAuth)
+
+  classOf[LedgerApiServer].getSimpleName when {
+    "client authorization is set to none" should {
```

Comment:
Wondering the same

Reply:
I did some research but haven't found much apart from these two sources:
https://netty.io/4.1/api/io/netty/handler/ssl/ClientAuth.html
https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLEngine.html#setWantClientAuth-boolean-

My understanding is that the `OPTIONAL` means that a server will ask for a client certificate and if not provided, continue TLS negotiations. If a certificate is provided, it must be valid - the connection will be rejected otherwise.
I've verified this interpretation by setting up a simple test.
I would say that the `OPTIONAL` doesn't give much value but that's a side note :)

Reply:
Could you please add a test for invalid certificate then?

Reply:
Done.